src/olympia/blocklist/models.py

```diff
@@ -485,7 +495,11 @@ def process_input_guids(
 
     def save_to_block_objects(self):
         assert self.is_submission_ready
-        assert self.action == self.ACTION_ADDCHANGE
+        assert self.action in (
+            self.ACTION_ADDCHANGE,
+            self.ACTION_HARDEN,
+            self.ACTION_SOFTEN,
+        )
```

Comment:
nit: if this was all new code I'd suggest using django-extended-choices so we could have all the actions defined together, and a sub-group for actions that need saving.  (marginal benefit with the code already written 🤷 )